### PR TITLE
Use resource instead of property map to store v2 config

### PR DESCRIPTION
**Pull request by 0xaptosj**
1. use resource instead of property map
2. add view to query func in config
3. update domain pricing to 
```
20 = 3 characters
10 = 4 characters
5 = 5 characters
1 = >= 6 characters
```

### CODE REVIEW

File: core_v2/sources/v2_config.move [hunk 1]
```diff
@@ -91,83 +112,89 @@ module aptos_names_v2::v2_config {
...
-        read_string(@aptos_names_v2, &config_key_tokendata_description())
+        borrow_global<Config>(get_config_signer_addr()).tokendata_description
     }
 
+    #[view]
     public fun tokendata_url_prefix(): String acquires Config {
-        read_string(@aptos_names_v2, &config_key_tokendata_url_prefix())
-    }
-
-    public fun domain_type(): String {
```
Comment (0xaptosj) [inline core_v2/sources/v2_config.move:137]: `domain_type` and `subdomain_type` are not used in v2

File: core_v2/sources/v2_config.move [hunk 2]
```diff
@@ -31,57 +30,79 @@ module aptos_names_v2::v2_config {
...
     const ENOT_AUTHORIZED: u64 = 1;
     /// Raised if there is an invalid value for a configuration
     const EINVALID_VALUE: u64 = 2;
+    /// Domain length is invalid, it must be at least 3 characters
+    const EINVALID_DOMAIN_LENGTH: u64 = 3;
 
     const SECONDS_PER_YEAR: u64 = 60 * 60 * 24 * 365;
+    const SECONDS_PER_DAY: u64 = 60 * 60 * 24;
 
     struct Config has key, store {
```
Comment (davidiw) [inline core_v2/sources/v2_config.move:46]: good to put this under some sort of resource group for all ANS

also this shouldn't have store

File: core_v2/sources/v2_config.move [hunk 3]
```diff
@@ -31,57 +30,79 @@ module aptos_names_v2::v2_config {
...
+        fund_destination_address: address,
+        max_number_of_seconds_registered: u64,
+        max_domain_length: u64,
+        min_domain_length: u64,
+        tokendata_description: String,
+        tokendata_url_prefix: String,
+        domain_price_length_3: u64,
+        domain_price_length_4: u64,
+        domain_price_length_5: u64,
+        domain_price_length_6_and_above: u64,
```
[the remark this answers is not in the document]
Comment (0xaptosj) [reply core_v2/sources/v2_config.move:58]: like hardcode it in `v2_price_model:: price_for_domain`? then we cannot change the price without redeploying contract, no?

File: core_v2/sources/v2_config.move [hunk 4]
```diff
@@ -31,57 +30,79 @@ module aptos_names_v2::v2_config {
...
-    ) acquires Config {
-        move_to(framework, Config {
-            config: property_map::empty(),
+    ) {
+        let constructor_ref = object::create_named_object(
+            framework,
+            CONFIG_OBJECT_SEED,
+        );
+        let extend_ref = object::generate_extend_ref(&constructor_ref);
+        let config_signer = object::generate_signer(&constructor_ref);
```
[the remark this answers is not in the document]
Comment (davidiw) [reply core_v2/sources/v2_config.move:75]: no, you just need resource groups

**Comment by 0xaptosj** [comment]
ready for review again

**Review by movekevin — APPROVED** [review]

**Pull request merged**